icu4c/source/i18n/unitsrouter.cpp

```diff
@@ -0,0 +1,65 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include <stdio.h>
+#include <utility>
+
+#include "cmemory.h"
+#include "cstring.h"
+#include "number_decimalquantity.h"
+#include "resource.h"
+#include "unitconverter.h" // for extractCompoundBaseUnit
+#include "unitsdata.h"     // for getUnitCategory
+#include "unitsrouter.h"
+#include "uresimp.h"
+
+U_NAMESPACE_BEGIN
+
+UnitsRouter::UnitsRouter(MeasureUnit inputUnit, StringPiece region, StringPiece usage,
+                         UErrorCode &status) {
+    // TODO: do we want to pass in ConversionRates and UnitPreferences instead
```

Comment:
There are pros and cons.  The main advantage of *not* making it global is that when you destroy the UnitsRouter, then all memory gets cleaned up. With a global, depending on how big the data is in memory, you may need to use complicated caching techniques to reduce residual memory usage.  If the data is small (less than a few KBs), then you can put it in a UInitOnce and just don't clean it up until someone calls `u_cleanup()`.